src/Stratis.Bitcoin.Features.MemoryPool/Rules/CheckAllInputsMempoolRule.cs

```diff
@@ -13,122 +14,130 @@ namespace Stratis.Bitcoin.Features.MemoryPool.Rules
     public class CheckAllInputsMempoolRule : MempoolRule
     {
         private readonly IConsensusRuleEngine consensusRuleEngine;
+        private readonly NodeDeployments nodeDeployments;
 
         public CheckAllInputsMempoolRule(Network network,
             ITxMempool mempool,
             MempoolSettings mempoolSettings,
             ChainIndexer chainIndexer,
             IConsensusRuleEngine consensusRuleEngine,
+            NodeDeployments nodeDeployments,
             ILoggerFactory loggerFactory) : base(network, mempool, mempoolSettings, chainIndexer, loggerFactory)
         {
             this.consensusRuleEngine = consensusRuleEngine;
+            this.nodeDeployments = nodeDeployments;
         }
 
+        /// <seealso>https://github.com/bitcoin/bitcoin/blob/febf3a856bcfb8fef2cb4ddcb8d1e0cab8a22580/src/validation.cpp#L770</seealso>
         public override void CheckTransaction(MempoolValidationContext context)
         {
+            // TODO: How should the RequireStandard setting interact with this?
             var scriptVerifyFlags = ScriptVerify.Standard;
-            if (!this.settings.RequireStandard)
-            {
-                // TODO: implement -promiscuousmempoolflags
-                // scriptVerifyFlags = GetArg("-promiscuousmempoolflags", scriptVerifyFlags);
-            }
 
-            // Check against previous transactions
+            // Check against previous transactions.
             // This is done last to help prevent CPU exhaustion denial-of-service attacks.
             var txdata = new PrecomputedTransactionData(context.Transaction);
             if (!this.CheckInputs(context, scriptVerifyFlags, txdata))
             {
-                // TODO: Implement Witness Code
-                //// SCRIPT_VERIFY_CLEANSTACK requires SCRIPT_VERIFY_WITNESS, so we
-                //// need to turn both off, and compare against just turning off CLEANSTACK
-                //// to see if the failure is specifically due to witness validation.
-                //if (!tx.HasWitness() && CheckInputs(Trx, state, view, true, scriptVerifyFlags & ~(SCRIPT_VERIFY_WITNESS | SCRIPT_VERIFY_CLEANSTACK), true, txdata) &&
-                //  !CheckInputs(tx, state, view, true, scriptVerifyFlags & ~SCRIPT_VERIFY_CLEANSTACK, true, txdata))
-                //{
-                //  // Only the witness is missing, so the transaction itself may be fine.
-                //  state.SetCorruptionPossible();
-                //}
+                // SCRIPT_VERIFY_CLEANSTACK requires SCRIPT_VERIFY_WITNESS, so we
+                // need to turn both off, and compare against just turning off CLEANSTACK
+                // to see if the failure is specifically due to witness validation.
+                if (!context.Transaction.HasWitness && this.CheckInputs(context, scriptVerifyFlags & ~(ScriptVerify.Witness | ScriptVerify.CleanStack), txdata) && !this.CheckInputs(context, scriptVerifyFlags & ~ScriptVerify.CleanStack, txdata))
```

Comment:
Is this still the same in core?

Reply:
Should be, the `seealso` linked code looks unchanged from the current master:

https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp#L917-L934